#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_hurd_detection.dpatch by Cyril Brulebois <cyril.brulebois@enst-bretagne.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix (workaround) POSIX incompatibility: unconditional use of PATH_MAX

@DPATCH@
diff -Nur qt4-x11-4.1.3~/src/3rdparty/dlcompat/dlfcn.h qt4-x11-4.1.3/src/3rdparty/dlcompat/dlfcn.h
--- qt4-x11-4.1.3~/src/3rdparty/dlcompat/dlfcn.h	2006-07-21 11:50:39.000000000 +0000
+++ qt4-x11-4.1.3/src/3rdparty/dlcompat/dlfcn.h	2006-07-21 11:59:52.000000000 +0000
@@ -99,4 +99,8 @@
 }
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 #endif /* _DLFCN_H_ */
diff -Nur qt4-x11-4.1.3~/src/corelib/io/qfsfileengine_unix.cpp qt4-x11-4.1.3/src/corelib/io/qfsfileengine_unix.cpp
--- qt4-x11-4.1.3~/src/corelib/io/qfsfileengine_unix.cpp	2006-07-21 11:50:38.000000000 +0000
+++ qt4-x11-4.1.3/src/corelib/io/qfsfileengine_unix.cpp	2006-07-21 11:59:52.000000000 +0000
@@ -40,6 +40,10 @@
 # include <private/qcore_mac_p.h>
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 void QFSFileEnginePrivate::init()
 {
 }
